packages/mazeppa/mazeppa.0.3.3/opam

```diff
@@ -0,0 +1,57 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "A modern supercompiler for call-by-value functional languages"
+maintainer: ["hirrolot <hirrolot@gmail.com>"]
+authors: ["hirrolot <hirrolot@gmail.com>"]
+license: "MIT"
+homepage: "https://github.com/mazeppa-dev/mazeppa"
+doc: "https://github.com/mazeppa-dev/mazeppa"
+bug-reports: "https://github.com/mazeppa-dev/mazeppa/issues"
+depends: [
+  "ocaml" {>= "5.1"}
+  "dune" {>= "3.15"}
+  "pprint"
+  "checked_oint" {>= "0.2.0"}
+  "ppx_deriving"
+  "ppx_string_interpolation"
+  "ppx_yojson_conv"
+  "ppx_blob" {>= "0.9.0"}
+  "bisect_ppx"
+  "menhir"
+  "spectrum" {>= "0.6.0"}
+  "clap"
+  "alcotest" {>= "1.7.0"}
+  "odoc" {with-doc}
+]
+available: arch != "arm32" & arch != "x86_32" & arch != "ppc64" & os != "win32" & switch != "5.3~alpha1" & os-family != "opensuse"
+x-ci-accept-failures: [
+  "fedora-38"
+  "fedora-39"
+  "fedora-40"
+  "oraclelinux-8"
+  "oraclelinux-9"
+  "freebsd"
```

Comment:
The failure on opensuse will keep appearing because it is not your package that is failing but a dependency:
```
#=== ERROR while compiling checked_oint.0.2.0 =================================#
# context              2.4.0~alpha1~dev | linux/x86_64 | ocaml-base-compiler.5.2.0 | file:///home/opam/opam-repository
# path                 ~/.opam/5.2/.opam-switch/build/checked_oint.0.2.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p checked_oint -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/checked_oint-7-362dbd.env
# output-file          ~/.opam/log/checked_oint-7-362dbd.out
### output ###
# File "lib/dune", lines 1-13, characters 0-311:
#  1 | (library
#  2 |  (public_name checked_oint)
#  3 |  (name checked_oint)
# ....
# 11 |   (names support)
# 12 |   (extra_deps int128_support.c)
# 13 |   (flags -O3)))
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlmklib -g -o lib/checked_oint_stubs lib/support.o)
# /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: lib/support.o: relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
# /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: failed to set dynamic section sizes: bad value
# collect2: error: ld returned 1 exit status
```

I am more concerned about all the other constraints on the compiler and the distributions, I'd like to see if they can be fixed or if there is an incompatibility that should be noted down

Reply:
It seems that `checked_oint` uses a constraint on older `opensuse`, which is why the CI tries to check whether a newer version works. I can re-release the `checked_oint` package with a fixed constraint on all `opensuse` distributions if the problem in [my comment](https://github.com/ocaml/opam-repository/pull/26662#issuecomment-2393637632) is not fixable.

Reply:
If the failure is due to a dependency, then I'd rather not constrain this package and let it fail. Once the depdendency is updated, if it fixes the issue, the revdeps will show if this package is then compatible or not with the other systems.

There is no need to re-release `checked_oint`, we can update the accept failure in the other package. And if it is relevant, we can add stricter constraints on the availability there.

Reply:
> Once the depdendency is updated, if it fixes the issue, the revdeps will show if this package is then compatible or not with the other systems.

But when `checked_oint` is updated to say `0.2.1`, wouldn't lower-bounds fail on `mazeppa`?

> There is no need to re-release `checked_oint`, we can update the accept failure in the other package.

Which package?

Reply:
> > Once the depdendency is updated, if it fixes the issue, the revdeps will show if this package is then compatible or not with the other systems.
> 
> But when `checked_oint` is updated to say `0.2.1`, wouldn't lower-bounds fail on `mazeppa`?

There is no lower bound check, and if it fails for a dependency is not the problem of mazeppa but of the dependency.

> > There is no need to re-release `checked_oint`, we can update the accept failure in the other package.
> 
> Which package?

checked_oint itself

Reply:
> checked_oint itself

How to update the accept failure in `checked_oint` then, just open a PR with an edit?

Reply:
Yes, thanks a lot!